### PR TITLE
DEPR Rename keyword "quantile" to "q" in `Rolling.quantile`

### LINKED ISSUE #52550: ENH: Differing variable name for quantile vs rolling.quantile
### Feature Type

- [ ] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

As of V2 in the code and in the documentation, the quantile method and the rolling quantile method take a different variable for the quantile to be calculated.
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.quantile.html
https://pandas.pydata.org/docs/reference/api/pandas.core.window.rolling.Rolling.quantile.html

where we have: 
q float for the former (quantile) and
quantile float for the latter (rolling.quantile).

I feel like this is unnecessary and provides a small mental overhead.
I propose that it should be changed to being consistent. Not sure which is better, perhaps q as used in quantile and is shorter but clear.

### Feature Description

def rolling.quantile(self,
      q,
     ....):

### Alternative Solutions

quantile = q

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Charlie-XIAO**
- [x] Closes #52550
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests)
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit)
- [x] Added an entry in the latest `doc/source/whatsnew/v2.1.0.rst` file

### CODE REVIEW

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -2382,8 +2385,8 @@ def kurt(self, numeric_only: bool = False):
         create_section_header("Parameters"),
         dedent(
             """
-        quantile : float
```
Comment (phofl) [inline pandas/core/window/rolling.py:2385]: Please keep the old name here and add a deprecation note that this will get renamed to q

Comment (phofl) [reply pandas/core/window/rolling.py:2385]: No need to add the new description already and also the cols thing is unnecessary

File: pandas/tests/window/test_rolling_quantile.py [hunk 2]
```diff
@@ -173,3 +173,11 @@ def test_center_reindex_frame(frame, q):
     )
     frame_rs = frame.rolling(window=25, center=True).quantile(q)
     tm.assert_frame_equal(frame_xp, frame_rs)
+
+
+def test_keyword_quantile_deprecated():
+    # GH #52550
+    # deprecate keyword 'quantile' of Rolling.quantile, rename as 'q'
```
Comment (phofl) [inline pandas/tests/window/test_rolling_quantile.py:180]: Can remove the comment here except the gh ref

File: pandas/core/window/rolling.py [hunk 3]
```diff
@@ -2384,6 +2387,9 @@ def kurt(self, numeric_only: bool = False):
             """
         quantile : float
             Quantile to compute. 0 <= quantile <= 1.
+
+            .. deprecated:: 2.1.0
+                This will be renamed to 'q'.
```
Comment (phofl) [inline pandas/core/window/rolling.py:2392]: Can you add "in a future version".

**Comment by mroeschke** [comment]
Could you deprecate this on `Expanding` too?

**Comment by mroeschke** [comment]
In rolling.py there's also this "private" quantile function

```
    def quantile(
        self,
        quantile: float,
        interpolation: QuantileInterpolation = "linear",
        numeric_only: bool = False,
    ):
```

could you change `quantile` to `q` here too?

**Comment by Charlie-XIAO** [comment]
> In rolling.py there's also this "private" quantile function, could you change `quantile` to `q` here too?

Sure, and since it is a private function, no deprecation is needed right?

**Comment by mroeschke** [comment]
> Sure, and since it is a private function, no deprecation is needed right?

Correct

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @Charlie-XIAO

**Pull request merged**